modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/JavaClientCodegen.java

```diff
@@ -812,6 +840,250 @@ public void setSerializationLibrary(String serializationLibrary) {
         }
     }
 
+    public void addOneOfNameExtension(Schema s, String name) {
+        ComposedSchema cs = (ComposedSchema) s;
+        if (cs.getOneOf() != null && cs.getOneOf().size() > 0) {
+            cs.addExtension("x-oneOfName", name);
```

Comment:
Is this `x-oneOfName` an extension taken for somewhere else, or one specific to our tool? If it is tooling specific, I think it will eventually need to be `x-oneOf-name` to match how other vendor extensions are named in https://github.com/OpenAPITools/openapi-generator/wiki/Vendor-Extensions. We'll need to re-evaluate our extensions and naming of these, because there doesn't seem to be an intuitive convention.

I created #4976 to track.

Reply:
It's specific to our tooling. I'll rename to `x-oneOf-name` and amend this PR.

Reply:
Done.